src/main/java/bio/terra/cli/service/SamService.java

```diff
@@ -285,26 +223,20 @@ public static GroupPolicy fromSamPolicy(String samPolicyName) {
    * @param userEmail email of the user to add
    */
   public void addUserToGroup(String groupName, GroupPolicy policy, String userEmail) {
-    GroupApi groupApi = new GroupApi(apiClient);
-    try {
-      // - try to add the email to the group
-      // - if this fails with a Bad Request error, it means the email is not found
-      // - so try to invite the user first, then retry adding them to the group
-      callAndHandleOneTimeErrorWithRetries(
-          () -> {
-            groupApi.addEmailToGroup(groupName, policy.getSamPolicy(), userEmail);
-            return null;
-          },
-          SamService::isBadRequest,
-          () -> {
-            inviteUserNoRetries(userEmail);
-            return null;
-          });
-    } catch (ApiException | InterruptedException ex) {
-      throw new SystemException("Error adding user to SAM group.", ex);
-    } finally {
-      closeConnectionPool();
-    }
+    // - try to add the email to the group
+    // - if this fails with a Bad Request error, it means the email is not found
+    // - so try to invite the user first, then retry adding them to the group
+    handleClientExceptions(
+        () ->
+            HttpUtils.callAndHandleOneTimeErrorWithRetries(
+                () ->
+                    new GroupApi(apiClient)
+                        .addEmailToGroup(groupName, policy.getSamPolicy(), userEmail),
```

Comment:
I'll keep this suggestion in mind for future improvements. But I think for now, it's not needed because I've wrapped all the calls to these `HttpUtils` methods in helpers for each client library that take care of their retryable arguments, so it's a smaller arg list in the many places this is actually called.